src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,31 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class BasePage {
+protected WebDriver driver;
+    protected   WebDriverWait wait;
+    protected   Actions actions;
+    public BasePage ( WebDriver givenDriver){
+        driver =givenDriver;
+        wait =new WebDriverWait(driver, Duration.ofSeconds(5));
+        actions = new Actions(driver);
```

Comment:
**Page Object Model and Page Factory**

![POM_vs_Factory_1](https://user-images.githubusercontent.com/104761482/236864030-34b89652-460d-4440-aa91-894332e90654.png)

![POM_vs_Factory_2](https://user-images.githubusercontent.com/104761482/236864098-d3ffe636-7ba9-484e-a72f-8b03ff024659.png)